pkgs/build-support/docker/default.nix

```diff
@@ -742,17 +720,23 @@ rec {
         # Use the temp folder we've been working on to create a new image.
         mv temp image/$layerID
 
+        # Add the new layer ID to the beginning of the layer list
+        (
+          # originally this used `sed -i "1i$layerID" layer-list`, but
+          # would fail if layer-list was completely empty.
+          echo "$layerID/layer.tar"
+          cat layer-list
+        ) | ${pkgs.moreutils}/bin/sponge layer-list
```

Comment:
That would break in the cross-compiling case.
Better to use:

```bash
(
 echo "$layerID/layer.tar"
 cat layer-list
) > layer-list-new
mv layer-list-new layer-list
```

This also removes the dependency on moreutils while having the same amount of external commands.

Reply:
Speaking of cross-compiling: All `buildInputs` above should become `nativeBuildInputs`